pkg/discovery/discovery.go

```diff
@@ -357,20 +355,38 @@ func updateStatus(client kubernetes.Interface, namespace string, status string,
 	return setPodStatusAnnotation(client, namespace, podStatus)
 }
 
-func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, pluginResultStatus string, pluginResultCounts map[string]int) error {
+func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, item results.Item) error {
 	podStatus, err := pluginaggregation.GetStatus(client, namespace)
 	if err != nil {
 		return errors.Wrap(err, "failed to get the existing status")
 	}
 
+	integrateResultsIntoStatus(podStatus, pluginType, &item)
+
+	return setPodStatusAnnotation(client, namespace, podStatus)
+}
+
+func integrateResultsIntoStatus(podStatus *pluginaggregation.Status, pluginName string, item *results.Item) {
 	for i := range podStatus.Plugins {
```

Comment:
Unrelated to this change, but the second value from `range` could be used here rather than needing to index into the slice with `i`.

Reply:
Only for the value lookups; the range loop makes a local copy and so sometimes you think a value is editable but it isn't necessary depending on if its a pointer/value/whatever. Most times if I'm editing the value I just err on the side of caution and use the index.

Testing this out, it is one of the cases where the value doesnt get edited when you try and set the status/counts a few lines below.

Reply:
Ah, I didn't know though that. TIL!